src/Components/Preview.lua

```diff
@@ -78,14 +102,19 @@ function Preview:updateDisplay()
 	if not target then
 		return
 	end
-	if self.expand then
+
+	if self.state.expand then
 		target.Parent = self.display
 	else
 		target.Parent = self.rootRef:getValue()
 	end
 end
 
-function Preview:refreshPreview()
+function Preview:refreshPreview(checkPreferenceToClear: boolean)
+	if checkPreferenceToClear and self.state.clearOutput then
+		LogService:ClearOutput()
+	end

```

Comment:
We don't need this boolean parameter. Put this in another function, or since it looks like it's only in one place, just put it there.